### PR TITLE
UIPFU-71: add `initialSelectedUsers` property

### PR DESCRIPTION
**Pull request by alisher-epam**
## JIRA ISUUE - [UIPFU-71](https://issues.folio.org/browse/UIPFU-71)
The feature is created based on this [Jira task](https://issues.folio.org/browse/UIU-2873)
As a user,
I want to see the assigned users list when opening the find users modal
so that I don't cause duplicate errors when assigning new users

Scenario:
- Given the user has already been assigned a users list
- When click the "Assign/unassign" button
- Then user should see the list of users already selected in the modal

Screenshot 2023-08-01 at 2 35 59 PM

### CODE REVIEW

**Review by usavkov-epam — APPROVED** [review]

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Review by Terala-Priyanka — CHANGES_REQUESTED** [review]
Could you add unit tests?

**Comment by alisher-epam** [comment]
@Terala-Priyanka I have added a unit test.

**Comment by alisher-epam** [comment]
@Terala-Priyanka I have merged the PR after adding unit tests since you are off today.

File: src/PluginFindUser.js [hunk 1]
```diff
@@ -123,6 +124,20 @@ PluginFindUser.propTypes = {
...
+      active: PropTypes.bool,
+      barcode: PropTypes.string,
+      personal: PropTypes.shape({
+        lastName: PropTypes.string,
+        firstName: PropTypes.string,
+        email: PropTypes.string,
+      }),
+      patronGroup: PropTypes.string,
+    }),
+  }),
```
Comment (zburke) [inline src/PluginFindUser.js:140]: Do we really need all this data? I expected this to just be an array of IDs.

**Review by zburke — COMMENTED** [review]
The tests are not quite right and need to be cleaned up before this can be merged. 

Also, the more I think about this, the more I wonder it feels like a much simpler solution would be just suck the user lists through set like `[...new Set(initial, selected)]` before making the API query. I guess that's not as nice since the modal won't reflect initial selected users, but it sure is simpler. Dunno; maybe worth considering though.

File: test/bigtest/tests/findUser-test.js [hunk 2]
```diff
@@ -154,6 +182,72 @@ describe('UI-plugin-find-user', function () {
...
+            email: PropTypes.string,
+          }),
+          patronGroup: PropTypes.string,
+        }),
+      }) },
+      { users: validProps },
+      'users',
+      'TestComponent'
+    );
+    expect(result).to.equal(undefined);
```
[the remark this answers is not in the document]
Comment (alisher-epam) [reply test/bigtest/tests/findUser-test.js:251]: Agree. I have not seen anyone writing to check prop validations as well. However, I have extracted the props inside `shapes.js` and the sonar showed 0% coverage and I was requested to write a test for resolving this issue.

File: test/bigtest/tests/findUser-test.js [hunk 3]
```diff
@@ -154,6 +182,72 @@ describe('UI-plugin-find-user', function () {
           expect(selectedUsers.length).to.equal(2);
         });
       });
+
+      describe('with initial selected users', function () {
+        it('returns selected users', function () {
+          expect(selectedUsers.length).to.equal(2);
+        });
+      });
```
[the remark this answers is not in the document]
Comment (alisher-epam) [reply test/bigtest/tests/findUser-test.js:190]: This is an optional prop and left it purposefully.

**Pull request merged**